### PR TITLE
Remove gulp

### PR DESCRIPTION
**Pull request by kuskoman**
+ Remove gulpfile and gulp related dependencies
+ Update prepare script and add build script
+ Generate typings in same directory as other files

Relates to #1

### CODE REVIEW

**Review by appellation — APPROVED** [review]

File: tsconfig.json [hunk 1]
```diff
@@ -1,15 +1,12 @@
...
-    "outDir": "./dist/",                     /* Redirect output structure to the directory. */
-    "strict": true,                           /* Enable all strict type-checking options. */
-    "esModuleInterop": true,                  /* Enables emit interoperability between CommonJS and ES Modules via creation of namespace objects for all imports. Implies 'allowSyntheticDefaultImports'. */
+    "target": "ES2017" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */,
+    "module": "commonjs" /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */,
+    "outDir": "./dist/" /* Redirect output structure to the directory. */,
+    "strict": true /* Enable all strict type-checking options. */,
+    "esModuleInterop": true /* Enables emit interoperability between CommonJS and ES Modules via creation of namespace objects for all imports. Implies 'allowSyntheticDefaultImports'. */,
     "declaration": true,
-    "declarationDir": "./dist/typings",
```
Comment (DevYukine) [inline tsconfig.json:9]: why was this removed?

File: tsconfig.json [hunk 2]
```diff
@@ -1,15 +1,12 @@
 {
   "compilerOptions": {
-    "target": "ES2017",                       /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
-    "module": "commonjs",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
-    "outDir": "./dist/",                     /* Redirect output structure to the directory. */
-    "strict": true,                           /* Enable all strict type-checking options. */
-    "esModuleInterop": true,                  /* Enables emit interoperability between CommonJS and ES Modules via creation of namespace objects for all imports. Implies 'allowSyntheticDefaultImports'. */
+    "target": "ES2017" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */,
+    "module": "commonjs" /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */,
+    "outDir": "./dist/" /* Redirect output structure to the directory. */,
```
Comment (DevYukine) [inline tsconfig.json:5]: why did you move the comma?

File: tsconfig.json [hunk 3]
```diff
@@ -1,15 +1,12 @@
...
+    "strict": true /* Enable all strict type-checking options. */,
```
Comment (DevYukine) [inline tsconfig.json:6]: same as above

**Review by DevYukine — CHANGES_REQUESTED** [review]
please do not edit the tsconfig and change styling for no reason in there

File: tsconfig.json (hunk 1 above)
Comment (kuskoman) [reply tsconfig.json:9]: Keeping typings folder for one file seems unreasonable, but ok, i can revert it back

File: tsconfig.json (hunk 2 above)
Comment (kuskoman) [reply tsconfig.json:5]: IDE auto formatting

**Pull request closed without merging**